src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -1222,6 +1222,31 @@ interface IInterface
         }
 
         [Fact]
+        public void BackCompatAsyncMain()
+        {
+            var source = @"
+using System.Threading.Tasks;
+
+class A
+{
+    async static Task Main(string[] args)
+    {
+        await Task.Factory.StartNew(() => { });
+    }
+    static void Main(){}
+}";
+
+            foreach (var langVersion in new LanguageVersion[] { LanguageVersion.CSharp7, LanguageVersion.CSharp7_1 })
+            {
+                var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1));
+                compilation.VerifyDiagnostics(
+                    // (6,23): error CS0017: Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point.
```

Comment:
>// (6,23): error CS0017: Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point. [](start = 20, length = 141)

This doesn't look like a backward compatible behavior. In 2.0.0.61501, I get a warning and the void main is used as an entry point:
```
(9,23): warning CS0028: 'A.Main(string[])' has the wrong signature to be an entry point
```